### DISCUSSION TITLE
Hierarchical Consensus Roadmap

### DISCUSSION THREAD

**Discussion by LefKok** [consensus-shipyard/consensuslab#99] [Project proposals]
As discussed on slack what we need to complete Hierarchical Consensus is the following. Let's add all resources in one place for easier visibility and access.

1. Cryptoeconomics: We are in direct contact with the CryptoEcon Lab
- Summary:https://hackmd.io/QA2ueIIWRB-JBZQNyHLJKg
- Plan: https://www.notion.so/Hierarchical-Consensus-a8811694060247eead5128e6972c6711 

2. Fraud/Validity proofs:
- Fraud Proofs: We can use classic fraud proofs relying on the censorship resistance of the parent chain, an assumption of a single honest validator (not eclipsed), and synchrony within the appeal window
- Validity Proofs with Lurk (https://drive.google.com/file/d/1oHAtWRAESUKUqi0z9AyLgwiqOcYMo3bE/view): Exploring collaboration. It will still need censorship resistance but only for liveness. The rest of the assumptions are lifted but Lurk needs a trusted setup (not 100% sure about this). Notes at https://hackmd.io/@consensuslab/HJDaWDEE5 

3. Robust Consensus: This is being part of Y3, especially #91, #98 

4.  Data Availability:
- Survey from CryptoNetLab: https://www.notion.so/pl-strflt/Data-availability-for-blockchain-17ce79eb09a84c5ca7ec4a994aa99646
- Attempt a more "permissioned solution" with @GuyGoren while at Novi: https://eprint.iacr.org/2022/455

5. Data Retrievability: Proposal and MVP from CryptoNetLab here (https://www.notion.so/pl-strflt/Data-Retrievability-Oracle-252f52f47b3944e8a54e8d0f553c5cd7#94890708571444df9b3d18336a3713ef)

**Comment by porcuquine** [comment]
> The rest of the assumptions are lifted but Lurk needs a trusted setup (not 100% sure about this).

In fact not. The initial Groth16 prototype would have required trusted setup, but we will plan to use Nova for this. Nova recursion is done and should be integrated in the near future. So there will be no trusted setup requirement for Lurk applications (unless for some very specific reason the application independently requires Groth16 or some other SNARK requiring the setup).
